librad/src/net/protocol.rs

```diff
@@ -461,9 +462,10 @@ where
 }
 
 #[async_trait]
-impl<S> GitStreamFactory for Protocol<S>
+impl<S, A> GitStreamFactory for Protocol<S, A>
 where
-    S: gossip::LocalStorage + 'static,
+    S: gossip::LocalStorage<Update = A> + 'static,
```

Comment:
It seems really strange to me that I need to put these trait bounds here -- I get that it's because the `impl` has them, but actually they could be just `forall`. What's the proper way to deal with this?

Reply:
Yeah, afaik the only way to deal with this is to have no bounds on the `impl`, and keep bounds on the methods only.